### DISCUSSION TITLE
What actions are processed using messenger queues?

### DISCUSSION THREAD

**Discussion by rjankiew** [pimcore/pimcore#11215] [Q&A / Support]
Hi, I have and issue with a large number of entries in messenger_messages table, that stacked up because of a period after 10.2 upgrade before any consumer was added. 

I am wondering what kind of actions are processed using messenger queues (pimcore_core, pimcore_maintenance)? I know of two actions that are delegated to the queue, and seem to make up the gist of it - indexing objects and generating previews. What are the other actions?

Let's say I would remove all entires in messenger_messages table, and run the indexing command. Could that lead to any issues in the system, if now added consumers will process any new entires?

**Comment by Fabian-Hofer-Mastertent** [comment]
Did you find any solution to this? I have the same problem (a lot of entries in the messenger_messages table)
The only solution I can think of at the moment is to start the consumer by hand and wait.
Sadly this is no solution for my situation (>1.000.000 entries) because it would take half a year to process :)

**Reply by dpfaffenbauer** [reply]
Most of the time it is quite save to truncate the table. Consequences are: Search might be out of date, and other maintenance task didnt run. Search can be run manually with `bin/console pimcore:search-backend-reindex, maintenance task run every day once anyway.

**Reply by Alanin** [reply]
When having a lot of messages in that table the queries are quite slow. I saw query time above 12 seconds. I also noticed that simultaneous queries (cause of multiple threads in supervisor) are blocking each other, because the query is made with '.... for update' which blocks that entry to be queried by another supervisor-thread. its waiting until that message is released.

are there any plans to make use of rabbitmq or another message bus system for core or even for enterprise bundles? we are using rabbitmq in our projects and it makes live much easier. also there is a large increase in performance when using rabbitmq...

we were able to route a few of the core messages to rabbitmq using the symfony configurations, but not all... some of the messages are hard coded using mysql tables

**Reply by Fabian-Hofer-Mastertent** [reply]
I ended up removing all messages with type "pimcore_core" directly in the database and let the "messenger:consume" the remaining messages with type "pimcore_maintainance" normally (because there where only a few left)

Thank you very much for your help.
